gamemodes/terrortown/entities/entities/ttt_c4/shared.lua

```diff
@@ -619,7 +633,10 @@ if SERVER then
 				return
 			elseif time < C4_MINIMUM_TIME or time > C4_MAXIMUM_TIME then
 				return
-			elseif IsValid(bomb:GetPhysicsObject()) and bomb:GetPhysicsObject():HasGameFlag(FVPHYSICS_PLAYER_HELD) then
+			elseif
+				IsValid(bomb:GetPhysicsObject())
```

Comment:
And why is this not in line with else if?

Reply:
this styling puts every literal on a new line so that it is easier to grasp what literals exist and which logical operators combine them.
This also separates the logical struct from its statements/conditions, which makes it easier to grasp that only the statement has changed and not the logical struct.

Reply:
imho it would look nicer like this:

```lua
if something then

elseif IsValid(bomb:GetPhysicsObject())
	and bomb:GetPhysicsObject():HasGameFlag(FVPHYSICS_PLAYER_HELD)
then return end
```

Reply:
```lua
if -- <-- logical control structure
	something() -- <-- expression
then -- <-- logical control structure
	do_something()
elseif -- <-- logical control structure
	IsValid(bomb:GetPhysicsObject()) -- <-- expression
	and bomb:GetPhysicsObject():HasGameFlag(FVPHYSICS_PLAYER_HELD) -- <-- expression
then -- <-- logical control structure
	return -- <-- expression
end -- <-- logical control structure
```

This way you don't mix the control structure of the if statement with the actual expressions that are used.
Not only do you get a more readable diff out of these bacause if the control structure doesn't change you won't have the control structure in the line diff (only as context), additionally having each literal on a new line starting with the logical operator allows you to easier grasp under which conditions a statement can terminate early and what is not evaluated in that case.


Without comments because github's syntax highlighting gets overly confused by comments ...
```lua
if
	something()
then
	do_something()
elseif
	IsValid(bomb:GetPhysicsObject())
	and bomb:GetPhysicsObject():HasGameFlag(FVPHYSICS_PLAYER_HELD)
then
	return
end
```

Reply:
I **really** dislike this structure because on first glance you can differntiate between something that is checked and something that is done when the check passes

Reply:
> allows you to easier grasp under which conditions a statement can terminate early and what is not evaluated in that case.

This is especially important for us because need so damn many nil checks ...

Reply:
> I _**really**_ dislike this structure because on first glance you can't differentiate between something that is checked and something that is done when the check passes

i think (?) i know what you mean, but instead of going to the start of the line going up to the first non-indented line is hardly more effort.

Reply:
but that's like defining a variable like this

```lua
local
	varname
	= value
```

it's a really weird way to write things

Reply:
I really like this style and dislike the harder to read inline style. 
Especially since this is only the case for multi line statements.

It is soo freaking hard to differentiate multiline conditions and the control structure of an `elseif` in the example above.

I like this consistent style much more and honestly i think it is something we get used too quite quickly.

Reply:
Honestly, I don't think Histaleks suggestion makes it easier to read, at all. 

This
```lua
if something() then
	do_something()
elseif IsValid(bomb:GetPhysicsObject())
	and bomb:GetPhysicsObject():HasGameFlag(FVPHYSICS_PLAYER_HELD)
then
	return
end
```

or
```lua
if something() then
	do_something()
elseif IsValid(bomb:GetPhysicsObject())
	and bomb:GetPhysicsObject():HasGameFlag(FVPHYSICS_PLAYER_HELD)
then return end
```
is so much cleaner looking

Reply:
I disagree, besides my points above those two examples have inconsistent `then` placements for no particular reasons other than "must use less lines". These make it worse to see at a glance what is "done" and not part of the condition.

Second point, the `elseif` is terrible to read, because the conditions don't start at the same indentation, which again makes it hard to grasp what is checked and what is done.

The style stylua uses, places the different "do" and "checked" parts into distinct groups separated by the keywords of the control structure.

I'm not advocating for a "cleaner" look, that's highly subjective. I'm advocating for a clearer and consistent code style that is next to impossible to misread and removes unnecessary noise in diffs. This will make reviewing changes much, much easier.

Reply:
Also i don't think we should try to argue about looks too much its much much more important that this is not a bad style and rather is something that is consistent and auto fixed.

Reply:
I guess I'm the only one who dislikes the new solution so the decision is on you